R/map_vis.R

```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
+  if(nrow(power_plant_list) == 0){
+    if(nrow(flood_control_dams) == 0){
+
+      ggplot(state_shape_trans)+
+        geom_sf(fill = "white")+
+        geom_sf(data = watersheds_city_trans, fill = "dodgerblue3", colour = "darkgrey") +
+        geom_sf(data = city_point,fill = "hotpink", colour = "black", size = 7, alpha = 0.7, shape = 22) +
+        theme(line = element_blank(),
+              legend.position = "none",
+              axis.text.x = element_blank(),
+              axis.text.y = element_blank(),
+              axis.title.x = element_blank(),
+              axis.title.y = element_blank(),
+              panel.background = element_blank(),
+              plot.title = element_text(hjust = 0.5)) +
+        ggtitle(paste0(city)) -> plot1
+
+      ggplot(df2) +
+        geom_raster(aes(x= Longitude, y=Latitude, fill=color)) +
+        scale_fill_identity() +
+        geom_sf(data = watersheds_city_trans, fill = "white",colour = "black", alpha = 0) +
+        theme(line = element_blank(),
+              legend.position = "none",
+              axis.text.x = element_blank(),
+              axis.text.y = element_blank(),
+              axis.title.x = element_blank(),
+              axis.title.y = element_blank(),
+              panel.background = element_blank(),
+              plot.title = element_text(hjust = 0.5)) -> plot2
+
+      all_data %>%
+        ggplot(aes(n_other_cities)) +
+        stat_ecdf(pad = F) -> cdf_plot
+      cdf_plot +
+        geom_point(data = layer_data(cdf_plot) %>% filter(x == number_cities),
+                   aes(x, y), col = "red", size = 5) +
+        xlab("Other Cities") +
+        ylab("") +
+        theme_classic() -> plot3
+
+      all_data %>%
+        ggplot(aes(irr_cons_BCM / yield_BCM)) +
+        stat_ecdf(pad = F) -> cdf_plot
+      cdf_plot +
+        geom_point(data = layer_data(cdf_plot) %>% filter(x == cons_per_yield),
+                   aes(x, y), col = "red", size = 5) +
+        xlab("IrrCons/Yield") +
+        ylab("") +
+        theme_classic() -> plot4
+
+      cowplot::plot_grid(plot3,plot4) -> graph_plot1
+
+      all_data %>%
+        ggplot(aes(n_hydro_plants + n_thermal_plants)) +
+        stat_ecdf(pad = F) -> cdf_plot
+      cdf_plot +
+        geom_point(data = layer_data(cdf_plot) %>% filter(x == total_plants),
+                   aes(x, y), col = "red", size = 5) +
+        xlab("Total Plants") +
+        ylab("") +
+        theme_classic() -> plot5
+
+      all_data %>%
+        ggplot(aes(cropland_fraction)) +
+        stat_ecdf(pad = F) -> cdf_plot
+      cdf_plot +
+        geom_point(data = layer_data(cdf_plot) %>% filter(x == crop_point),
+                   aes(x, y), col = "red", size = 5) +
+        xlab("Crop Fraction") +
+        ylab("") +
+        theme_classic() -> plot6
+      cowplot::plot_grid(plot5,plot6) -> graph_plot2
+
+      cowplot::plot_grid(plot1,plot2,graph_plot1,graph_plot2)
```

Comment:
Suggest re-ordering: Other cities, Number of power plants, Cropland fraction, Crop water consumption (as % of yield)